app/src/src_nonfreenet/org/breezyweather/sources/pagasa/PagasaResultConverter.kt

```diff
@@ -0,0 +1,278 @@
+                            total = it.precipitation?.attributes?.value?.toDoubleOrNull()
+                        ),
+                        wind = Wind(
+                            degree = it.windDirection?.attributes?.deg?.toDoubleOrNull(),
+                            speed = it.windSpeed?.attributes?.mps?.toDoubleOrNull()
+                        ),
+                        relativeHumidity = it.relativeHumidity?.attributes?.value?.toDoubleOrNull()
+                    )
+                )
+            }
+        }
+    }
+    return hourlyList
+}
+
+private fun getWindDirection(
+    direction: String?,
+): Double? {
+    return when (direction) {
+        "N" -> 0.0
+        "NNE" -> 22.5
+        "NE" -> 45.0
+        "ENE" -> 67.5
+        "E" -> 90.0
+        "ESE" -> 112.5
+        "SE" -> 135.0
+        "SSE" -> 157.5
+        "S" -> 180.0
+        "SSW" -> 202.5
+        "SW" -> 225.0
+        "WSW" -> 247.5
+        "W" -> 270.0
+        "WNW" -> 292.5
+        "NW" -> 315.0
+        "NNW" -> 337.5
+        else -> null
+    }
+}
+
+private fun getHourlyWeatherText(
+    context: Context,
+    symbol: String?,
+): String? {
+    return symbol?.let {
+        with(symbol) {
+            when {
+                startsWith("01") -> context.getString(R.string.common_weather_text_clear_sky)
+                startsWith("02") -> context.getString(R.string.common_weather_text_mainly_clear)
+                startsWith("03") -> context.getString(R.string.common_weather_text_partly_cloudy)
+                startsWith("04") -> context.getString(R.string.common_weather_text_cloudy)
+                startsWith("05") -> context.getString(R.string.common_weather_text_rain_showers)
+                startsWith("06") -> "Thunder showers" // thundershower
+                startsWith("07") -> context.getString(R.string.common_weather_text_rain_snow_mixed_showers)
+                startsWith("08") -> context.getString(R.string.common_weather_text_snow_showers)
+                startsWith("09") -> context.getString(R.string.common_weather_text_rain)
+                startsWith("10") -> context.getString(R.string.common_weather_text_rain_heavy)
+                startsWith("11") -> context.getString(R.string.weather_kind_thunderstorm) // heavy thunderstorm
+                startsWith("12") -> context.getString(R.string.common_weather_text_rain_snow_mixed)
+                startsWith("13") -> context.getString(R.string.common_weather_text_snow)
+                startsWith("14") -> "Thunder snowstorm" // thunder snow
+                startsWith("15") -> context.getString(R.string.common_weather_text_fog)
+                startsWith("16") -> "Thunder sleet shower" // thunder sleet shower
+                startsWith("17") -> "Thunder snow shower" // thunder snow shower
+                startsWith("18") -> context.getString(R.string.weather_kind_thunderstorm) // thunderstorm
+                startsWith("19") -> "Thunder rain and snow mixed" // thunder sleet
```

Comment:
I'll take a look, but since Philippines is a tropical country, I don't think `16`, `17` or `19` will be used ever. ;) We can consider that a low priority, so don't let the missing resource strings stop you from merging.